pkg/autohealing/cloudprovider/openstack/provider.go

```diff
@@ -262,6 +296,66 @@ func (provider OpenStackCloudProvider) Repair(nodes []healthcheck.NodeInfo) erro
 	return nil
 }
 
+// UpdateHealthStatus can update the cluster health status by
+func (provider OpenStackCloudProvider) UpdateHealthStatus(masters []healthcheck.NodeInfo, workers []healthcheck.NodeInfo) error {
+	log.Infof("Start to update cluster health status.")
+	clusterName := provider.Config.ClusterName
+
+	healthStatus := "UNHEALTHY"
+	healthStatusReasonMap := make(map[string]string)
+	healthStatusReasonMap["updated_at"] = time.Now().String()
+
+	if len(masters) == 0 && len(workers) == 0 {
+		// No unhealthy node passed in means the cluster is healthy
+		healthStatus = "HEALTHY"
+		healthStatusReasonMap["api"] = "ok"
+		healthStatusReasonMap["nodes"] = "ok"
+	} else {
+		healthStatus = "UNHEALTHY"
+		if len(workers) > 0 {
+			for _, n := range workers {
+				// TODO: Need to figure out a way to reflect the detailed error information
+				healthStatusReasonMap[n.KubeNode.Name+"."+n.FailedCheck] = "error"
+			}
+		} else {
+			// TODO: Need to figure out a way to reflect detailed error information
+			healthStatusReasonMap["api"] = "error"
+		}
+	}
+
+	jsonDumps, err := json.Marshal(healthStatusReasonMap)
+	if err != nil {
+		return fmt.Errorf("Failed to build health status reason for cluster %s, error: %v", clusterName, err)
```

Comment:
Error string should not be capitalized, so s/Failed/failed, please change others as well.

Reply:
No problem. Will do.